service/controller/v14/resource/endpoint/delete.go

```diff
@@ -20,11 +21,13 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 	// times. We do not want to delete the whole endpoint only because one pod is
 	// gone. We only delete the whole endpoint when it does not contain any IP
 	// anymore. Removing IPs is done on update events.
-	if endpointToDelete != nil && isEmptyEndpoint(*endpointToDelete) {
+	if isEmptyEndpoint(*endpointToDelete) {
```

Comment:
This is the point here. We should always deal with a structure. In case we do not want to delete the endpoint, we want to log about it anyways. Right now there is an issue that the code panics because of the added logging from an earlier PR, because in some cases there was `nil` dispatched.